src/invidious/routes/video_playback.cr

```diff
@@ -277,7 +277,11 @@ module Invidious::Routes::VideoPlayback
       return error_template(500, ex)
     end
 
-    fmt = video.fmt_stream.find(nil) { |f| f["itag"].as_i == itag } || video.adaptive_fmts.find(nil) { |f| f["itag"].as_i == itag }
+    if itag.nil?
+      fmt = video.fmt_stream[-1]
```

Comment:
```suggestion
      fmt = video.fmt_stream[-1]?
```

for reference:
 - https://play.crystal-lang.org/#/r/epbz (causes an IndexError Exception)
 - https://play.crystal-lang.org/#/r/epc0 (return nil)

Reply:
That's in the event where a video doesn't have a progressive HTTP itag available (e.g in progress or recently ended livestreams)

Reply:
That's a good idea, returning a 404 if the list is empty. Thanks!